modules/swagger-jaxrs2/src/main/java/io/swagger/jaxrs2/OperationParser.java

```diff
@@ -32,256 +35,497 @@
+            return Optional.empty();
+        }
+        getContents(requestBody.content()).ifPresent(requestBodyObject::setContent);
+        return Optional.of(requestBodyObject);
+    }
+
+    public static Optional<ApiResponses> getApiResponses(final io.swagger.oas.annotations.responses.ApiResponse[] responses) {
+        if (responses == null) {
+            return Optional.empty();
+        }
+        ApiResponses apiResponsesObject = new ApiResponses();
+        for (io.swagger.oas.annotations.responses.ApiResponse response : responses) {
+            ApiResponse apiResponseObject = new ApiResponse();
+            getContent(response.content()).ifPresent(apiResponseObject::content);
+            if (StringUtils.isNotBlank(response.description())) {
+                apiResponseObject.setDescription(response.description());
+                getLinks(response.links()).ifPresent(apiResponseObject::setLinks);
+            }
+            if (StringUtils.isNotBlank(response.responseCode())) {
+                apiResponsesObject.addApiResponse(response.responseCode(), apiResponseObject);
+            } else {
+                apiResponsesObject.addApiResponse(RESPONSE_DEFAULT, apiResponseObject);
+            }
+        }
+
+        return Optional.of(apiResponsesObject);
+    }
+
+    public static Optional<Content> getContents(io.swagger.oas.annotations.media.Content[] contents) {
+        if (contents == null) {
+            return Optional.empty();
+        }
+        Content contentObject = new Content();
+        MediaType mediaType = new MediaType();
+        for (io.swagger.oas.annotations.media.Content content : contents) {
+            ExampleObject[] examples = content.examples();
+            for (ExampleObject example : examples) {
+                getMediaType(mediaType, example).ifPresent(mediaTypeObject -> contentObject.addMediaType(content.mediaType(), mediaType));
+            }
+        }
+        if (contentObject.size() == 0) {
+            return Optional.empty();
+        }
+        return Optional.of(contentObject);
+    }
+
+    public static Optional<Content> getContent(io.swagger.oas.annotations.media.Content annotationContent) {
+        if (annotationContent == null) {
+            return Optional.empty();
+        }
+
+        Content content = new Content();
+        MediaType mediaType = new MediaType();
+        Class<?> schemaImplementation = annotationContent.schema().implementation();
+        Map<String, Schema> schemaMap;
+        if (schemaImplementation != Void.class) {
+            schemaMap = ModelConverters.getInstance().readAll(schemaImplementation);
+            schemaMap.forEach((k, v) -> mediaType.setSchema(v));
+        } else {
+            getSchemaFromAnnotation(annotationContent.schema()).ifPresent(mediaType::setSchema);
+        }
+        if (StringUtils.isNotBlank(annotationContent.mediaType())) {
+            content.addMediaType(annotationContent.mediaType(), mediaType);
```

Comment:
For some reason, the schemas are being added in-line.  They need to be added to the `components` object and set as a reference.  This:

```
"content": {
  "application/json": {
    "schema": {
      "type": "object",
      "properties": {
        "id": {
          "type": "integer",
          "format": "int64"
        },
        "username": {
          "type": "string"
        },
        "firstName": {
          "type": "string"
        },
        "lastName": {
          "type": "string"
        },
        "email": {
          "type": "string"
        },
        "password": {
          "type": "string"
        },
        "phone": {
          "type": "string"
        },
        "userStatus": {
          "type": "integer",
          "description": "User Status",
          "format": "int32",
          "enum": [
            null
          ]
        }
      }
    }
  }
}
```

Should be this:

```
"content": {
  "application/json": {
    "schema": {
      "$ref": "#/components/schemas/User"
    }
  }
}
```

With the corresponding components:

```
"components": {
  "schemas": {
    "User": {
      "type": "object",
      "properties": {
        "id": {
          "type": "integer",
          "format": "int64"
        },
        "username": {
          "type": "string"
        },
        "firstName": {
          "type": "string"
        },
        "lastName": {
          "type": "string"
        },
        "email": {
          "type": "string"
        },
        "password": {
          "type": "string"
        },
        "phone": {
          "type": "string"
        },
        "userStatus": {
          "type": "integer",
          "description": "User Status",
          "format": "int32",
          "enum": [
            null
          ]
        }
      }
    }
  }
}
```